### PR TITLE
[EICNET-2610]fix: Fix solr indexing issues due to null group/file

### PR DESCRIPTION
**Pull request by Dromadoros**
How to test it:

- [x] You can just take DB from QA and try reindexing everything, verify that no errors are shown

### CODE REVIEW

File: lib/modules/eic_groups/modules/eic_share_content/src/Service/ShareManager.php [hunk 1]
```diff
@@ -222,13 +222,16 @@ public function isShared(NodeInterface $node, GroupInterface $target_group = NUL
...
+    if ($target_group instanceof GroupInterface) {
+      $group_content_type = $this->defineGroupContentType($target_group);
+      $query->condition('type', $group_content_type, 'LIKE');
       $query->condition('gid', $target_group->id());
     }
+
+    $query->condition('entity_id', $node->id());
+
     return $this->entityTypeManager->getStorage('group_content')->loadMultiple($query->execute());
   }
```
Comment (4hardfire) [inline lib/modules/eic_groups/modules/eic_share_content/src/Service/ShareManager.php:236]: I believe I did some improvements in here -> https://github.com/EIC-EA/EIC-community-D8/pull/1748/files. Can you have a look?

**Review by 4hardfire — CHANGES_REQUESTED** [review]

File: lib/modules/eic_groups/modules/eic_share_content/src/Service/ShareManager.php (hunk 1 above)
Comment (Dromadoros) [reply lib/modules/eic_groups/modules/eic_share_content/src/Service/ShareManager.php:236]: I can remove this change but we still need the other

Comment (4hardfire) [reply lib/modules/eic_groups/modules/eic_share_content/src/Service/ShareManager.php]: I resolved the conflicts with develop branch.

**Review by 4hardfire — APPROVED** [review]
This is ok!

**Pull request merged**